### ISSUE TITLE
Output probabilities are wrong due

### ISSUE THREAD

**Issue by vjung99** [DLS5-Omics/multimolecule#7]
# Problem statement
First of all, thank you for the development of this library, it has been very useful in my research. 

I have noticed that the output probabilities of some models are off compared with their original implementation, and I have found 2 reasons for it:

1) Biases in the MLMHead decoder are not copied from the original implementation, at least in some of the models I have seen. 
2) New tokens introduced by the standardised vocabularies that are not in the original vocabularies get an assigned embedding that is the average of the original vocabulary's embeddings. This means that they have logits that come into the softmax calculation and thus lower the probabilities over the original vocabulary. 

# Example
For example, for SpliceBERT: 

## 1. Output logits are not the same because of MLMHead decoder biases 
 ```
with torch.device("cuda:0"), torch.no_grad():
    # SPLICEBERT_PATH is the path to the SpliceBERT.1024nt model as downloaded from the original repository
    sp_og = AutoModelForMaskedLM.from_pretrained(SPLICEBERT_PATH)
    tok_og = AutoTokenizer.from_pretrained(SPLICEBERT_PATH)
    sp_mm = SpliceBertForMaskedLM.from_pretrained("multimolecule/splicebert")
    tok_mm = AutoTokenizer.from_pretrained("multimolecule/splicebert")
    sp_og.to("cuda:0")
    sp_mm.to("cuda:0")
    sp_og.eval()
    sp_mm.eval()
    input_og = "T A G C [MASK] T A T C A G A C T G A T G T T A"
    input_mm = input_og.replace("[MASK]", "<mask>").replace(" ", "").replace("T", "U")
    torch.manual_seed(42)
    torch.cuda.manual_seed(42)
    np.random.seed(42)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False

    # Disable flash attention for the original SpliceBERT model
    for i in range(len(sp_mm.splicebert.encoder.layer)):
        sp_mm.splicebert.encoder.layer[i].attention.self.flash = False

    tokenized_og = tok_og(input_og, return_tensors="pt")
    tokenized_mm = tok_mm(input_mm, return_tensors="pt")

    out_og = sp_og(**tokenized_og)
    out_mm = sp_mm(**tokenized_mm)
 ```

We can check that the output representation is the same (or at least very close): 

```
torch.allclose(out_og["hidden_states"][-1],out_mm["hidden_states"][-1], atol=1e-6)
> True
```

But the logits for the token "A" for the masked position are not the same for example: 
```
a_idx_og = tok_og.convert_tokens_to_ids("A")
a_idx_mm = tok_mm.convert_tokens_to_ids("A")

#Original implementation, mm implementation
out_og["logits"][0][5][a_idx_og].item(), out_mm["logits"][0][5][a_idx_mm].item()
> (0.8041356801986694, 0.794284462928772)
```

### Fix
Replacing the biases in sp_mm.lm_head.decoder.bias by the ones in sp_og.cls.predictions.decoder.bias fixes this : 
```
# adding after instantiating models in the code block above
og_to_mm = [0, 3, 1, 2, 4, 10, 6, 7, 8, 9] # original token mapping to mm model index
sp_mm.lm_head.decoder.bias[og_to_mm] = sp_og.cls.predictions.decoder.bias
out_og = sp_og(**tokenized_og)
out_mm = sp_mm(**tokenized_mm)
```
Leads to:
```
#Original implementation, mm implementation
out_og["logits"][0][5][a_idx_og].item(), out_mm["logits"][0][5][a_idx_mm].item()
> (0.8041356801986694, 0.8041356801986694)
```

## 2. New tokens change output probabilities
Since softmax is applied over the logits, the fact that there are new tokens introduced whose embeddings are the average of the original vocabulary embeddings according to : https://github.com/DLS5-Omics/multimolecule/blob/4f77e733822cab20336d230016e6554d46adadfc/multimolecule/tokenisers/utils.py#L69 

means that there are new logits who come into the softmax calculation, leading to the wrong probability distribution over the tokens. For example:
```
out_og = sp_og(**tokenized_og)
out_mm = sp_mm(**tokenized_mm)

#Original implementation, mm implementation
out_og["logits"][0].softmax(-1)[5][a_idx_og].item(), out_mm["logits"][0].softmax(-1)[5][a_idx_mm].item()
> (0.40978527069091797, 0.09764634072780609)
```

### Fix
A straightforward fix for this would be to set the bias in the decoder of the MLMHead of the model to -inf. This means that for the new tokens, the logits would be -inf and thus not affect the softmax calculation:

```
vocab_idx_not_in_og = list(set(range(tok_mm.vocab_size)).difference(set(og_to_mm)))
sp_mm.lm_head.decoder.bias[vocab_idx_not_in_og] = torch.tensor(float("-inf"))
out_og = sp_og(**tokenized_og)
out_mm = sp_mm(**tokenized_mm)

#Original implementation, mm implementation
out_og["logits"][0].softmax(-1)[5][a_idx_og].item(), out_mm["logits"][0].softmax(-1)[5][a_idx_mm].item()
> (0.40978527069091797, 0.40978527069091797)
```
